src/main/java/hng_java_boilerplate/product/controller/ProductController.java

```diff
@@ -61,4 +62,44 @@ public ResponseEntity<ProductSearchDTO> searchProducts(
         productSearchDTO.setSuccess(true);
         return new ResponseEntity<>(productSearchDTO, HttpStatus.OK);
     }
+
+
+    @GetMapping("/{productId}/stock")
+    public ResponseEntity<ProductStockDTO> getProductStock(@PathVariable String productId) {
+        Product product = productService.getProductStock(productId);
+        ProductStockDTO productStockDTO = new ProductStockDTO(
+                product.getId(),
+                product.getCurrentStock(),
+                product.getLastUpdated()
+        );
+        return new ResponseEntity<>(productStockDTO, HttpStatus.OK);
+    }
+
+    @PostMapping
+    @PreAuthorize("hasRole('ROLE_USER')")
```

Comment:
The role based access here gives only the user with user role the authorization to create a product and an admin will not be able to create a product. Is this the expected behaviour?